src/ptvsd/adapter/messages.py

```diff
@@ -407,13 +425,24 @@ def initialized_event(self, event):
         # also remove the 'initialized' event sent from IDE messages.
         pass
 
+    @_only_allowed_while("initializing")
+    def process_event(self, event):
+        try:
+            debuggee.parse_pid(event)
+        except Exception:
+            # If we couldn't retrieve or validate PID, we can't safely continue
```

Comment:
Shouldn't this be the case only for launch? for attach it should not really matter.

Reply:
This particular bit will work as intended, because `parse_pid()` won't do anything, so it won't raise an exception.

However, the bit in `_configure()` that does `wait_for_pid()` is wrong for this reason, and breaks "attach".

Reply:
I'll put this in a conditional for clarity tho.